WordPress/src/test/java/org/wordpress/android/ui/sitecreation/progress/SiteCreationProgressViewModelTest.kt

```diff
@@ -145,13 +147,30 @@ class SiteCreationProgressViewModelTest : BaseUnitTest() {
     }
 
     @Test
-    fun `on retry click emits service event with the previous result`() {
+    fun `on retry click after service error emits service event with the previous result`() = test {
         startViewModel()
         viewModel.onSiteCreationServiceStateUpdated(SERVICE_ERROR)
         viewModel.retry()
         assertEquals(viewModel.startCreateSiteService.value?.previousState, SERVICE_ERROR.payload)
     }
 
+    @Test
+    fun `on retry click after network error restarts site creation service`() = test {
+        whenever(networkUtils.isNetworkAvailable()).thenReturn(false, true)
+        startViewModel()
+        advanceUntilIdle()
+        viewModel.retry()
+        verify(startServiceObserver).onChanged(any())
+    }
+
+    @Test
+    fun `on retry click after cart error retries to create cart`() = testWith(CART_ERROR) {
+        startViewModel(SITE_CREATION_STATE.copy(domain = PAID_DOMAIN))
+        viewModel.onSiteCreationServiceStateUpdated(SERVICE_SUCCESS)
+        viewModel.retry()
+        verify(createCartUseCase, times(2)).execute(any(), any(), any(), any(), eq(false))
+    }
```

Comment:
I was getting failures on 5 tests, and an exception on this one, until I realized I still had the patch applied :man_facepalming: . After reverting the patch, everything is working as expected. :relieved:  Thanks for adding tests!

Reply:
That troubled me as well for a while when writing the test 🤣 !
Thank you 🙇🏻